### PR TITLE
Allow replacing actual call to perform HTTP request via subclassing

### PR DESCRIPTION
**Pull request by fake-name**
Basically, I have my own urllib wrapper that I'm maintaining almost entirely out of spite. It has it's own cookie/UA/header/etc... management, and I'd like to be able to just wrap that instead of having to move things back and forth between it and the requests session continuously. 

This change basically moves the actual calls to the parent  `super().request()` call into a stub function, so I can subclass `CloudScraper()`, and then just replace the body of `perform_request()` with my own HTTP fetching machinery. 

I'm not sure this is something of interest to really anyone other then myself, but it's also a really simple change (and could potentially be useful for testing purposes/mocking as well). Being able to plug in any arbitrary HTTP(s) transport seems like a nice feature too.

### CODE REVIEW

**Comment by VeNoMouS** [comment]
Whats wrong with the pre/post hooking?

**Comment by VeNoMouS** [comment]
Can you change to the dev branch please.

**Comment by VeNoMouS** [comment]
```
This change basically moves the actual calls to the parent super().request() call into a stub function, so I can subclass CloudScraper(), and then just replace the body of perform_request() with my own HTTP fetching machinery.
```

TL;DR: **monkey patch**.. 😄 

But yea change it to dev branch and ill commit

**Comment by fake-name** [comment]
> TL;DR: monkey patch.. 😄

I would, if there was a straightforward way to do so. I suppose I could try to clobber `requests.sessions.Session` *before* it's imported by cloudscraper, but that sounds like a easy way to wind up with import-order driven behaviour that'll fail in weird ways when you're not looking. 

> But yea change it to dev branch and ill commit

Oh whoops, derp.

**Comment by fake-name** [comment]
Woot!

**Comment by VeNoMouS** [comment]
:) thanks for contributing

**Comment by fake-name** [comment]
Thanks for dealing with cloudflare so I don't have to!

**Comment by VeNoMouS** [comment]
It'll go through on next production release, unless some else happens ... it'll prob be when cloudflare changes something in ~6 days
